### ISSUE TITLE
BUG: `vak.files.spect.find_audio_fname` raises ValueError when given a Windows path

### ISSUE THREAD

**Issue by avanikop** [NickleDave/vak#431]
Hi!

I am using vak0.4.0b6. When I run the predict step I get the following error:

> Traceback (most recent call last):
  File "D:\anacondnavi\envs\vak040b6\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "D:\anacondnavi\envs\vak040b6\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "D:\anacondnavi\envs\vak040b6\Scripts\vak.exe\__main__.py", line 7, in <module>
  File "D:\anacondnavi\envs\vak040b6\lib\site-packages\vak\__main__.py", line 45, in main
    cli.cli(command=args.command, config_file=args.configfile)
  File "D:\anacondnavi\envs\vak040b6\lib\site-packages\vak\cli\cli.py", line 30, in cli
    COMMAND_FUNCTION_MAP[command](toml_path=config_file)
  File "D:\anacondnavi\envs\vak040b6\lib\site-packages\vak\cli\predict.py", line 42, in predict
    core.predict(
  File "D:\anacondnavi\envs\vak040b6\lib\site-packages\vak\core\predict.py", line 276, in predict
    audio_fname = files.spect.find_audio_fname(spect_path)
  File "D:\anacondnavi\envs\vak040b6\lib\site-packages\vak\files\spect.py", line 54, in find_audio_fname
    raise ValueError(
ValueError: unable to determine filename of audio file from: C:\Users\avanikoparkar\OneDrive - UT Cloud\Documents\analysis\tolabel\bu38o58\01Mar2017_screen\010317\spectrograms_generated_220103_122409\bu38o58_010317_162314.3.cbin.spect.npz

I have checked the audiofile (cbin) using evsonganaly and it seems to be a song file of normal size. Checked the cbin.spect.npz file contents as well.

**Comment by NickleDave** [comment]
Hi @avanikop sorry you're having this issue.

What's happening here is that `predict` calls the function `vak.files.spect.find_audio_fname` to try and determine the name of the audio file from the path to the spectrogram. So I don't think it's related to the files themselves. It's something about how that function handles the filename.

Looks like the path is: `C:\Users\avanikoparkar\OneDrive - UT Cloud\Documents\analysis\tolabel\bu38o58\01Mar2017_screen\010317\spectrograms_generated_220103_122409\bu38o58_010317_162314.3.cbin.spect.npz`

The way this works is that `files.spect.find_audio_fname` removes extensions from the file name until it finds one that is a valid audio format (currently, 'cbin' or 'wav'). That function calls another, `vak.files.files.find_fname`, that uses a [regex](https://docs.python.org/3/howto/regex.html) to try and find one of the valid format extensions in the path.

Is there anything that's different from how you usually run `vak predict`? I'm trying to think of what could be causing the issue.

The first thing that jumps out at me is that it's a Windows path and the path has spaces in it, because of the cloud drive.  
Are you maybe running from a different directory than usual that causes these extra parts of the path to get added?  
Not to make it sound like it's your fault, just trying to figure out why we haven't run into this before.  

Also is this the very first audio file in the dataset? I.e, the first row in the .csv that `vak prep` generated? Or does it work on some other rows first but then it crashes when it gets to this one?

My guess is that something about the path as a string (spaces?) breaks the regex.  
This is probably a good reason to switch to using `pathlib` here instead of using a regex, but we need to confirm that's the source of the issue. I will see if I can reproduce locally.

**Comment by avanikop** [comment]
You're right, I'm running it in a different path than my previous birds. I tried running it outside the OneDrive-UT folder, and it works! Thanks!

**Comment by NickleDave** [comment]
Hi again @avanikop

Yes I think it's just the Windows path. You were on Mac OS before, right? Just trying to confirm this could be the source of the issue.

When I copy the path from above and try to make a variable from it as a string, I get the following error:
```
In [1]: path = 'C:\Users\avanikoparkar\OneDrive - UT Cloud\Documents\analysis\tolabel\bu38o58\01Mar2017_screen\010317\spectrograms_generated_220103_122409\bu38o58_010317_162314.3.
   ...: cbin.spect.npz'
  File "<ipython-input-1-1b347089a01c>", line 1
    path = 'C:\Users\avanikoparkar\OneDrive - UT Cloud\Documents\analysis\tolabel\bu38o58\01Mar2017_screen\010317\spectrograms_generated_220103_122409\bu38o58_010317_162314.3.cbin.spect.npz'
           ^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 2-3: truncated \UXXXXXXXX escape

```

Translating from error-ese:  
https://stackoverflow.com/questions/1347791/unicode-error-unicodeescape-codec-cant-decode-bytes-cannot-open-text-file
> Typical error on Windows because the default user directory is C:\user\<your_user>, so when you want to pass this path as a string argument into a Python function, you get a Unicode error, just because the \u is a Unicode escape. If the next 8 characters after the \u are not numeric this produces an error.

So it's reading `'\U'` as the Unicode escape.

I think something like this is happening inside the `from_fname` function?

> To solve it, just double the backslashes: C:\\user\\<\your_user>... This will ensure that Python treats the single backslashes as single backslashes.

If you need a workaround right this very instant, you could fix this by manually, although it's a bit annoying.  
Basically you use `pathlib` to convert all the strings in the dataset csv. `pathlib` knows how to handle converting Windows paths into strings that won't cause errors in Python.

I can reply here with a code snippet that will do that, as a workaround.  
But I'm realizing that I'm a little confused about what the paths look like in your dataset .csv so I'm not quite sure how to write the snippet.

Could you please attach the dataset .csv that's causing the problem? (You'll probably have to hide it in a .zip first so GitHub will let you attach it).

I will figure out a long-term fix in the next couple of days, which will probably involve re-writing these functions to use `pathlib`.  
It's good you caught this since it's a fragile bit of the code, thank you for spotting the bug.

**Comment by NickleDave** [comment]
> You're right, I'm running it in a different path than my previous birds. I tried running it outside the OneDrive-UT folder, and it works! Thanks!

Huh. Ok.
That was a simple fix :laughing:  
Ignore whatever I said above :innocent: :stuck_out_tongue_closed_eyes: 

I will close this for now but open a related issue to rewrite these functions to use `pathlib` anyway
